blivet/devices/md.py

```diff
@@ -521,6 +523,40 @@ def pre_commit_fixup(self, current_fmt=False):
         if getattr(self.format, "mountpoint", None) == "/boot/efi":
             self.metadata_version = "1.0"
 
+    def _remove_stale_lvm(self):
+        """ Remove any stale LVM metadata that pre-existed in a new array's on-disk footprint.
+
+            XXX This should only be run from _post_create.
+        """
+        import time
+        log.debug("waiting 5s for activation of stale lvm on new md array %s", self.path)
+        time.sleep(5)
+        udev.settle()
+        info = udev.get_device(self.sysfs_path)
+        fmt = get_format(udev.device_get_format(info), device=self.path, exists=True)
+        if fmt.type != "lvmpv":
+            return
+
+        pvs_info.drop_cache()
+        pv_info = pvs_info.cache.get(self.path)
+        if pv_info is None:
+            return
+
+        vg_name = pv_info.vg_name
+        vg_uuid = pv_info.vg_uuid
+        if vg_name and vg_uuid:
+            log.info("removing stale LVM metadata found on %s", self.name)
+            try:
+                # As of libblockdev-2.15 the vg name is passed straight to
+                # com.redhat.lvmdbus1.Manager.LookUpByLvmId, which returns
+                # a DBus object path based on vg name or UUID without the
+                # caller having to specify which is being passed.
```

Comment:
This unfortunately doesn't work with the "non-dbus" plugin. I think the right way to do this should be to use the extra args and pass the `--select vg_uuid=XXXX` option (so something like this -- https://gist.github.com/vojtechtrefny/782b0160471e341635a3a24467791fe3). The problem is that this doesn't work with the dbus plugin because it doesn't expect name to be NULL/None.

We either need to change the dbus plugin to extract the uuid in the [remove method](https://github.com/tasleson/lvm-dubstep/blob/master/lvmdbus/vg.py#L189) from the extra arguments, or change the "old" plugin to accept uuid instead of name (without using the extra arguments).

Reply:
Just FYI, changing *libblockdev* to accept UUIDs in place of PV/VG/LV paths/names has been a goal for a long time. I just haven't gotten to it yet.